src/builder/definitions/base_class_def.py

```diff
@@ -1,16 +1,37 @@
 import inspect
 import typing
+from abc import abstractmethod
 
 from stubmaker.builder.common import Node, BaseRepresentationsTreeBuilder, BaseDefinition, get_annotations
 from typing_inspect import is_generic_type, get_generic_bases
 
 
+def _try_patch_pydantic_init_signature(cls):
+    """Pydantic implements custom class signature but assigns it only to cls.__signature__ but not
+    cls.__init__.__signature__ due to technical reasons: https://github.com/pydantic/pydantic/issues/1032. This
+    function tries to detect if the class is pydantic BaseModel and manually assigns signature to __init__.__signature__
+    """
+
+    try:
+        from pydantic import BaseModel
+        if issubclass(cls, BaseModel):
+            cls.__init__.__signature__ = cls.__signature__.replace(
+                parameters=[
+                    inspect.Parameter(name='self', kind=inspect.Parameter.POSITIONAL_OR_KEYWORD),
+                    *cls.__signature__.parameters.values(),
```

Comment:
It will fail with
```python
ValueError: wrong parameter order: positional or keyword parameter before positional-only parameter
```
if there will be `inspect.Parameter .POSINIONAL_ONLY` args after self